os/kernel/irq/irq_csection.c

```diff
@@ -123,7 +123,7 @@ static bool irq_waitlock(int cpu)
 	while (spin_trylock_wo_note(&g_cpu_irqlock) == SP_LOCKED) {
 		/* Is a pause request pending? */
 
-		if (up_cpu_pausereq(cpu)) {
+		if (up_cpu_pausereq(cpu) || up_get_gating_flag_status(cpu) == 1) {
```

Comment:
This will break the standardisation for kernel codes, it's difficult to understand what does flag status = 1 correlates to, it is too chipset specific (ie. for RTL8730E flash operation).
Some countermeasure should be taken to address this issue later.

Reply:
You are right. But since the code in the kernel folder is interacting with arch code, i think it is necessary to call low level apis. As of now, I am not sure how we can change this. Maybe in future, if we add more chipsets, then we can handle this in a better way.